pkg/platform/common/vmlayer/cloudlet.go

```diff
@@ -89,21 +89,38 @@ func (v *VMPlatform) GetPlatformNodes(cloudlet *edgeproto.Cloudlet) []NodeInfo {
 	return nodes
 }
 
-// GetCloudletImageToUse decides what image to use based on
+// GetCloudletImageName decides what image to use based on
 // 1) if MEX_OS_IMAGE is specified in properties and not default, use that
 // 2) Use image specified on startup based on cloudlet config
-func (v *VMPlatform) GetCloudletImageToUse(ctx context.Context, updateCallback edgeproto.CacheUpdateCallback) (string, error) {
+func (v *VMPlatform) GetCloudletImageName(ctx context.Context) (string, string, error) {
 	imgFromProps := v.VMProperties.GetCloudletOSImage()
 	if imgFromProps != "" {
 		log.SpanLog(ctx, log.DebugLevelInfra, "using image from MEX_OS_IMAGE property", "imgFromProps", imgFromProps)
-		return imgFromProps, nil
+		return imgFromProps, "", nil
 	}
 
 	vmBaseImage := v.VMProperties.CommonPf.PlatformConfig.CloudletVMImagePath
 	if vmBaseImage == "" {
-		return "", fmt.Errorf("Get cloudlet image failed, cloudletVMImagePath not set")
+		return "", "", fmt.Errorf("Get cloudlet image failed, cloudletVMImagePath not set")
 	}
 	imageNameWithoutExt := util.RemoveExtension(filepath.Base(vmBaseImage))
+	return imageNameWithoutExt, vmBaseImage, nil
+}
+
+// GetCloudletImageToUse decides what image to use based on
+// 1) if MEX_OS_IMAGE is specified in properties and not default, use that
+// 2) Use image specified on startup based on cloudlet config
+// 3) Add image to cloudlet image storage if not
+func (v *VMPlatform) GetCloudletImageToUse(ctx context.Context, updateCallback edgeproto.CacheUpdateCallback) (string, error) {
+	imageNameWithoutExt, vmBaseImage, err := v.GetCloudletImageName(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	// means we use image from props
+	if vmBaseImage == "" {
+		return imageNameWithoutExt, nil
+	}
```

Comment:
You're right I was confused for a while why we need two strings. But now I think I get it. I suggest you rename `vmBaseImage` to `imagePath`. The distinction (which wasn't clear right away) is that one is just the name, and the other is the full path. Then I suggest changing this check to:
```
// if no image path, nothing to download, so just return the image name
if imagePath == "" {
    return imageNameWithoutExt, nil
}
```